src/Pos/Launcher/Scenario.hs

```diff
@@ -72,11 +73,18 @@ runNode' NodeResources {..} workers' plugins' = ActionSpec $ \vI sendActions ->
                         ", pk hash: "%build) pk addr pkHash
 
     -- Synchronously join the Kademlia network before doing any more.
-    -- If we can't join the network, an exception is raised and the program
-    -- stops.
-    case topologyRunKademlia (ncTopology (ncNetworkConfig nrContext)) of
-        Nothing -> return ()
-        Just kInst -> kademliaJoinNetwork kInst (kdiInitialPeers kInst)
+    --
+    -- In case of end-user p2p modes, if we can't join the network, an exception
+    -- is raised and the program stops.
+    -- But for relay nodes, not finding any peers in the network is not fatal.
+    -- It could be that this is the first relay coming up. The relay can still
+    -- do its work using its static topology.
+    case ncTopology (ncNetworkConfig nrContext) of
+        TopologyCore _ (Just kInst) -> kademliaJoinNetworkNoThrow kInst (kdiInitialPeers kInst)
+        TopologyRelay _ (Just kInst) -> kademliaJoinNetworkNoThrow kInst (kdiInitialPeers kInst)
+        TopologyP2P _ _ kInst -> kademliaJoinNetwork kInst (kdiInitialPeers kInst)
+        TopologyTraditional _ _ kInst -> kademliaJoinNetwork kInst (kdiInitialPeers kInst)
```

Comment:
This is another one of those policy things. Can we pull this out as a function that tells us if:
 1. we need to join the kademlia network and fail if we cannot
 2. we need to try to join the kademlia network but sit there passively if we cannot join
 3. don't need to join the kademlia network at all.